protocols/gossipsub/src/behaviour.rs

```diff
@@ -3159,14 +3196,23 @@ where
             sender: Sender::new(self.config.connection_handler_queue_len()),
             topics: Default::default(),
             dont_send: LinkedHashMap::new(),
+            extensions: None,
         });
         // Add the new connection
         connected_peer.connections.push(connection_id);
+        let receiver = connected_peer.sender.new_receiver();
+
+        if connected_peer.connections.len() <= 1 {
+            // If this is the first connection send extensions message.
+            self.send_message(
+                peer_id,
+                RpcOut::Extensions(Extensions {
+                    test_extension: Some(true),
+                }),
+            );
+        }
```

Comment:
I don't think we want to do this here. 

This will send an extension message to all peers that connect to us, even those that don't support extensions. So I think we'd be sending invalid messages to peers on the network that don't support extensions. 

Once we create a handler its first message back to us should be the kind of peer that this peer is. We should use this entry point for this code I think, this way it will still be the first message we send (need to make sure it goes before subscriptions) and then we can make sure we send it to only peers that support it. 

This is the same for inbound connections.